### DISCUSSION TITLE
How to Configure Location used for Redirects

### DISCUSSION THREAD

**Discussion by KentonParton** [awslabs/aws-lambda-web-adapter#465] [Q&A]
Hi all,

I have the following setup:

my-application.example.com --> CloudFront --> Lambda (aws-lambda-web-adapter - FastAPI)

The FastAPI application has a route `/playground`. When I open `my-application.example.com/playground` in the browser, it redirects to the lambda URL `https://xxx.lambda-url.us-east-1.on.aws/playground/` instead of the Route53 URL. I suspect this is because CloudFront is forwarding a `Location` of the Lambda URL instead of `my-application.example.com`.

Is there a way to override this behaviour in aws-lambda-web-adapter or will I need to update `Location` using Lambda@Edge?

Thanks!

**Comment by KentonParton** [comment]
Hi @bnusunny, is it possible to achieve the above? Appreciate your help!

**Reply by bnusunny** [reply]
Sorry, I missed this question. I will take a look in a few days.

**Comment by bnusunny** [comment]
This is not a Lambda Web Adapter issue. It's generally how FastAPI works behind a proxy or CDN. Here is how to get it working: 

1. Add a custom hear to CloudFront Distribution's origin. For example: `X-Forwarded-Host=my-application.example.com`
2. Add a middleware in FastAPI to rewrite `Location` in the response based on the custom header's value. Here is an example. 

```python
from fastapi import FastAPI, Request
from starlette.middleware.base import BaseHTTPMiddleware
from starlette.types import ASGIApp
from urllib.parse import urlparse, urlunparse


class RewriteRedirectMiddleware(BaseHTTPMiddleware):
    def __init__(self, app: ASGIApp):
        super().__init__(app)

    async def dispatch(self, request: Request, call_next):
        response = await call_next(request)

        if 'Location' in response.headers:
            original_location = response.headers['Location']
            forwarded_host = request.headers.get('X-Forwarded-Host')

            if forwarded_host:
                parsed_url = urlparse(original_location)
                new_url = urlunparse(parsed_url._replace(netloc=forwarded_host))
                response.headers['Location'] = new_url

        return response


app = FastAPI()
app.add_middleware(RewriteRedirectMiddleware)


@app.get("/")
async def root():
    return {"headers": "Hello world!"}


@app.get("/playground/")
async def playground(request: Request):
    return {"headers": request.headers}

```

**Comment by bnusunny** [comment]
Lambda Function URL does not support custom domain name. It requires the `Host` header to match its own URL.
